src/modules/delivery/delivery.service.ts

```diff
@@ -0,0 +1,32 @@
+import { HttpService } from '@nestjs/axios';
+import { Injectable, BadRequestException } from '@nestjs/common';
+
+@Injectable()
+export class CorreiosService {
+  constructor(private readonly httpService: HttpService) {}
+
+  async calculateDeliveryTime(cep: string): Promise<number> {
+    try {
+      const response = await this.httpService
+        .get(`https://api.correios.com.br/${cep}`)
+        .toPromise();
+
+      const prazoEntrega = response.data.prazoEntrega;
+
+      if (!prazoEntrega) {
+        throw new BadRequestException(
+          'Endereço inválido. Por favor, insira um endereço válido.',
+        );
+      }
+
+      return parseInt(prazoEntrega);
+    } catch (error) {
+      if (error.response && error.response.status === 404) {
```

Comment:
Aqui você teria que utilizar a função que criei "handleError"
Porque dentro dele tem a seguinte lógica:
Se for uma exceção que você mandou em código como você fez no throw new BadRequestException
Ele vai relançar pro cliente, mas se for uma exceção não tratada vai lançar um internal server error com uma mensagem padrão